### PR TITLE
Auto-label

### LINKED ISSUE #1368: [FR] Filter processing
Add the possibility to perform action according to rules: 

Filter "Postgres" : 
  - Postgres in title or content

Perform : 
 - Mark as read
 - Assign tag "postgres"

### LINKED ISSUE #2380: Auto tag For the new entry
I want to give new entry some user tags by the keyword for the entry. 
I want the tag list showed in the page.
Have any good way ?

### LINKED ISSUE #2420: Tag all articles from feed
Hello,

As it seems to be difficult to implement multi categories due to compatibility with API, it would be great to have the possibility to set tags on feed. All articles from a feed would have feed tags.

It would allow to make filtering easier. For example I'm using rss-bridge for Twitter and right now (without tweak) I can't properly hide Twitter feeds from a category. If I could add them a "Twitter" tag, then I would be able to make a filter with : `-#Twitter` that hide them all. 

(Right now to achieve the same functionality I made a custom Twitter Bridge that set "author" to Twitter as I don't use it, and then I'm able to use `-author:Twitter` ; but it's more a workaround that a proper solution.)

Thank you :)

### LINKED ISSUE #3279: Assign labels automatically based on article title, content, and tags
Hello,

Is it possible to assign labels automatically based on title, content, and tags?

For example: News article with 'Games' in the title > Assign it automatically to label 'Games'.
I think this feature is really important. I can't organize my RSS feeds without this.

### LINKED ISSUE #5599: [Feature] Automatic assignment of labels during feed refresh
**Is your feature request related to a problem? Please describe.**
Often people want to be informed quickly about a certain topic. Until now, you always had to use the search, possibly with saved searches.

**Describe the solution you’d like**
It would be much more comfortable if there was a way to automatically assign labels during feed refresh. Possibly with the same syntax that the search already offers now.

**Describe alternatives you’ve considered**
An alternative would be to display the saved searches in the menu on the left. However, I think it could be resource-intensive to run all searches again and again to display e.g. the number of found items in the menu.

### LINKED ISSUE #5728: Can the function of adding tags be enhanced?
When refreshing the source, if a predefined word is found in the title or content, then automatically assign a specific tag attribute to this piece of news. This would greatly facilitate management and categorization when dealing with a large number of news articles.

### PR DESCRIPTION
**Pull request by Alkarex**
Add labels automatically to incoming articles
fix https://github.com/FreshRSS/FreshRSS/issues/2380
fix https://github.com/FreshRSS/FreshRSS/issues/2420
fix https://github.com/FreshRSS/FreshRSS/issues/3279
fix https://github.com/FreshRSS/FreshRSS/discussions/4947
fix https://github.com/FreshRSS/FreshRSS/issues/5728
fix https://github.com/FreshRSS/FreshRSS/issues/5599

image

image

### CODE REVIEW

**Comment by Nebucatnetzer** [comment]
Managed to get it working.
For some reason I wasn’t able to create a label called “news” and I couldn’t find an error for it. “my-news” worked fine however. Going to test it some more.

**Comment by Alkarex** [comment]
> or some reason I wasn’t able to create a label called “news”

Maybe you already have a category called "news"?

**Comment by Nebucatnetzer** [comment]
A right yes I do. I didn’t realise that the labels can’t be the same as the normal categories.
Other than that it behaves exactly like I expected and it allows me to create automatic “searches” so that I see instantly if there is an article that interests me.👍

**Comment by Alkarex** [comment]
Thanks for the feedback 👍🏻

File: app/Controllers/tagController.php [hunk 1]
```diff
@@ -84,6 +83,51 @@ public function deleteAction(): void {
 		}
 	}
 
+
+	/**
+	 * This action updates the given tag.
+	 */
+	public function updateAction(): void {
```
Comment (Inverle) [inline app/Controllers/tagController.php:90]: #7768

**Pull request merged**